tests/integration/test_store.py

```diff
@@ -1366,6 +1374,15 @@ def test_span_extraction(
 
     relay.send_event(project_id, event)
 
+    if discard_transaction:
+        transactions_consumer.poll(timeout=2.0) is None
+
+        # We do not accidentally produce to the events topic:
+        assert events_consumer.poll(timeout=2.0) is None
```

Comment:
Without the extra `has_event` checks, the envelope processor would serialize an empty event that gets published to the events topic.